lib/cldr/export/data/numbers.rb

```diff
@@ -7,108 +7,181 @@ class Numbers < Base
+          if aliased
+            return xpath_to_symbols_alias(aliased["path"])
+          end
 
-        def symbols
-          select("numbers/symbols[@numberSystem=\"latn\"]/*").each_with_object({}) do |node, result|
+          select("numbers/symbols[@numberSystem=\"#{number_system}\"]/*").each_with_object({}) do |node, result|
             result[name(node).to_sym] = node.content
           end
         end
 
-        def format(type)
-          result = select("numbers/#{type}Formats/#{type}FormatLength").each_with_object({}) do |format_length_node, format_result|
-            format_nodes = select(format_length_node, "#{type}Format")
+        def format(number_system, type)
+          number_system_node = select_single("numbers/#{type}Formats[@numberSystem=\"#{number_system}\"]")
+          return {} unless number_system_node
 
-            format_key = format_length_node.attribute("type")
-            format_key = format_key ? format_key.value : :default
+          aliased = select_single(number_system_node, "alias")
+          if aliased
+            return xpath_to_format_alias(aliased["path"], type)
+          end
 
-            if !format_nodes.empty?
-              format_nodes.each do |format_node|
-                format_result[format_key] ||= select(format_node, "pattern").each_with_object({}) do |pattern_node, pattern_result|
-                  pattern_key_node = pattern_node.attribute("type")
+          result = select("numbers/#{type}Formats[@numberSystem=\"#{number_system}\"]/#{type}FormatLength").each_with_object({}) do |format_length_node, format_result|
+            format_length_key = format_length_node["type"]&.to_sym || default_format_length_type
 
-                  pattern_count_node = pattern_node.attribute("count")
+            aliased = select_single(format_length_node, "alias")
+            if aliased
+              format_result[format_length_key] = xpath_to_redirect(aliased["path"], number_system, type)
+              next
+            end
 
-                  pattern_key = pattern_key_node ? pattern_key_node.value : :default
+            format_result[format_length_key] = if format_length_key == default_format_length_type
+              parse_default_currency_format_length_node(number_system, format_length_node, type)
+            else
+              parse_currency_format_length_node(format_length_node, type)
+            end
+          end
 
-                  if pattern_count_node
-                    pattern_count = pattern_count_node.value
+          result
+        end
 
-                    if pattern_result[pattern_key].nil?
-                      pattern_result[pattern_key] ||= {}
-                    elsif !pattern_result[pattern_key].is_a?(Hash)
-                      raise "can't parse patterns with and without 'count' attribute in the same section"
-                    end
+        def parse_default_currency_format_length_node(number_system, format_length_node, type)
```

Comment:
Suggest taking `currency` out of the method name since it can be multiple types.

```suggestion
        def parse_default_format_length_node(number_system, format_length_node, type)
```